### DISCUSSION TITLE
Probaby easy to fix - deployment build issue

### DISCUSSION THREAD

**Discussion by uk-medic-coder** [vercel/community#5765] [Help] [answered]
### Summary

Hi, I'm new to this, and have got a NextJS app working on localhost and now wish to deploy it to Vercel (hobby plan).

First I uploaded to github:

```
git add .
git commit -m "Initial commit"
```

Then connected Vercel - but I got build errors (below)

I noticed that /node-modules were not uploaded by git to github. Also .env.local, which has my environment variables in, was not uploaded to github.

![2024-02-07_14-23](https://github.com/vercel/community/assets/117088291/f9aaf165-4f09-4e49-9804-560e71fdfedf)

![2024-02-07_14-30](https://github.com/vercel/community/assets/117088291/999ec95a-0e14-44b3-86bd-a39c22133bbb)

### Steps to Reproduce

(as above)

**Comment by amyegan** [comment]
Hey @uk-medic-coder! 

When you create a git commit the `node_modules` directory is typically left out. The `package.json` file should include a list of dependencies to install, and that list will be used to install the right packages when the project is being deployed. So no worries there.

`.env` files are also typically left out because they may contain sensitive values that you wouldn't want exposed in a GitHub repo. [Environment Variables](https://vercel.com/docs/projects/environment-variables#) can be added to Project settings instead. You can upload your current `.env` file or copy and paste its contents in the environment variable input form.

It looks like you have a Next.js project, so you should make sure the `Next.js` framework preset is selected in [Build and Development Settings](https://vercel.com/docs/deployments/configure-a-build#build-and-development-settings). That will set the appropriate commands to use when automatically deploying the app.

I wasn't able to replicate the errors with a fresh `create-next-app` example repo, but I suspect that's because your app is a bit more complex than just the starter example. If you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), someone here may be able to recognize what's going wrong and suggest a more specific solution.

**Comment by uk-medic-coder** [comment]
OK, thanks for the help :) My github project is set to private.

My package.json:

```
{
  "name": "csttracker",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "export": "next export",
    "lint": "next lint"
  },
  "dependencies": {
    "@hookform/resolvers": "^3.3.4",
    "axios": "^1.6.7",
    "clsx": "^2.1.0",
    "exceljs": "^4.4.0",
    "next": "14.1.0",
    "react": "^18",
    "react-dom": "^18",
    "react-hook-form": "^7.50.1",
    "yup": "^1.3.3"
  },
  "devDependencies": {
    "@types/react": "18.2.52",
    "eslint": "^8",
    "eslint-config-next": "14.1.0"
  }
}
```

and the project is very simple:

![image](https://github.com/vercel/community/assets/117088291/b30e4ad2-c5a8-40b8-a0ba-176927df74dd)

**Reply by uk-medic-coder** [reply]
An example file: ajax.jsx:

```
import axios, { AxiosResponse } from 'axios';

import * as gnrl from '@common/general'

// ==============================================

export function fetchAjaxData(methodd, urll, dataa)
{
    return new Promise(function(resolve, reject)
    {
        const configuration = {
            method: methodd,
            url: urll,
            withCredentials: true,      // needed to stop CORS errors
            data: dataa
        }

        axios(configuration)
        .then(function (response) {
            resolve(response);
        })
        .catch(function (error) {
            gnrl.processErrorObject(error, "Axios > fetchAjaxData");
            reject();
        });
    });
}
```

**Reply by uk-medic-coder** [reply, marked as answer]
I managed to solve it!

When I ran `npm run build` on my local machine, I had some errors in the JSX which caused the build to fail.

When I fixed these, the build ran fine locally. I reuploaded all files to github, and the app built fine in vercel :)
